lib/Sema/CSSimplify.cpp

```diff
@@ -8988,6 +9120,12 @@ ConstraintSystem::simplifyApplicableFnConstraint(
       type1, origType2, subflags, locator);
 
   if (shouldAttemptFixes() && result == SolutionKind::Error) {
+    // Skip those fixes in case we are only attempting to check if it's possible
+    // to insert an explicit call in an anchor expr that produces a callable
+    // value.
+    if (locator.isForImplicitApplyCallableValue())
```

Comment:
If I understand it correctly, all of the checks for `isForImplicitApplyCallableValue` sprinkled around the code are necessary because there is no way to come back to a generic "fix" for a type mismatch?... If that's the same then the solution would be to add one more option to the overload set for explicit call which would just register that generic fix and do nothing else - that way we don't need any special handling for call overloads which didn't match...

Reply:
Humm, that would be to add a fixed choice constraint as the last constraint before [`addOverloadSet`](https://github.com/apple/swift/blob/789ada59f316f842d525980970b23c91b64b3f48/lib/Sema/CSSimplify.cpp#L7157)?

Reply:
Yeah, we'd probably need to do a lookup as part of a fix and adjust the overload set to have that "fallback" choice.

Reply:
Hum, when trying to do this I was struggling because as my understanding we would have to `Constraint::createFixedChoice` but didn't know what would be the "fallback" overload choice? 

> do a lookup as part of a fix 

Sorry, not sure I understand this part, that means we have to do a lookup in order to create the fallback fixed choice constraint?

Reply:
As I see it the logic here would be to:

- Use `performMemberLookup` to form a member set, add a "fallback" choice like "don't do anything but apply a fix (e.g. `ContextualMismatch` one)
- Construct an `ApplicableFunction` constraint to imitate presence of a missing call.

This way if any of the choices found by lookup produce other fixes they'd be considered a worse solution vs. the "fallback".

Reply:
But at this point, before [addOverloadSet](https://github.com/apple/swift/blob/789ada59f316f842d525980970b23c91b64b3f48/lib/Sema/CSSimplify.cpp#L7157) we already have a member set right? Since a member lookup is performed right before it ... or do you mean we should do when repairing the mismatch instead of `simplifyApplicableFnConstraint`? 
Sorry for all the questions, I'll probably have to give think a bit more about to understand the idea of this approach :)

Reply:
bq. do you mean we should do when repairing the mismatch instead of simplifyApplicableFnConstraint 

That's exactly what I mean, I we should perform lookup and add additional choices while forming a disjunction.

Reply:
Ahh great, thank you @xedin :) 
I will get back to this soon

Reply:
Hey @xedin :) 
While getting back to this I've found this behavior [SR-13844](https://bugs.swift.org/browse/SR-13844), so I'm thinking before moving on with this I should take a look at that problem. WDYT? 
I'm just asking because I've seen that there is a TODO of some refactors needed in `simplifyDynamicCallableApplication` and I'm trying to organize between some SRs/PRs I've opened.

Reply:
@LucianoPAlmeida I think it would make sense to refactor some of the code around that area before going back to this, it might make it easier.

Reply:
Right, thanks @xedin! 
I'll take a look at that then

Reply:
Great!